sled-agent/src/illumos/running_zone.rs

```diff
@@ -310,14 +311,18 @@ impl RunningZone {
 
 impl Drop for RunningZone {
     fn drop(&mut self) {
-        match Zones::halt_and_remove_logged(&self.inner.log, self.name()) {
-            Ok(()) => {
-                info!(self.inner.log, "Stopped and uninstalled zone")
-            }
-            Err(e) => {
-                warn!(self.inner.log, "Failed to stop zone: {}", e)
+        let log = self.inner.log.clone();
+        let name = self.name().to_string();
+        tokio::task::spawn(async move {
+            match Zones::halt_and_remove_logged(&log, &name).await {
+                Ok(()) => {
+                    info!(log, "Stopped and uninstalled zone")
+                }
+                Err(e) => {
+                    warn!(log, "Failed to stop zone: {}", e)
+                }
             }
-        }
+        });
```

Comment:
This is a minor divergence in behavior - *most* of the "sync -> async" calls eventually bubbled up to an asynchronous function, but not here, because it's in a `Drop` impl. Instead, we spawn a task to do the zone termination.

This *does* mean that the deletion of the zone may be deferred until after the `RunningZone` is dropped.